ext/gd/config.m4

```diff
@@ -149,7 +149,9 @@ dnl display a warning but eventually return normally, making a simple link
 dnl or run test insufficient.
 AC_DEFUN([PHP_GD_CHECK_FORMAT],[
   old_LIBS="${LIBS}"
-  LIBS="${LIBS} ${GD_SHARED_LIBADD}"
+  LIBS="${LIBS} ${GDLIB_LIBS}"
```

Comment:
```suggestion
  LIBS="${LIBS} ${GD_SHARED_LIBADD}"
```

I'd still use the cleaner `GD_SHARED_LIBADD` here like before. It worked previously on many installations without issues.

`GDLIB_LIBS` might be raw `-L/path/to/some/lib/ -lgd`

and 

`GD_SHARED_LIBADD` will be transformed into `-Wl,-rpath,/path/to/some/lib -L/path/to/some/lib -lgd` when building shared gd extension (--enable-gd=shared --with-external-gd).

When building static gd extension (--enable-gd --with-external-gd) the LIBS and LDFLAGS already contain proper library locations from the PHP_EVAL_LIBLINE.
For example:
LIBS=... -lm -lxml2 -lsqlite3 -lxml2 -lgd
LDFLAGS=-Wl,-rpath,/path/to/some/lib -L/path/to/some/lib

And the `GDLIB_CFLAGS` is perfect change for this as added.

Reply:
> --enable-gd=shared

Today is the day I learned about this option :facepalm:

I'm glad you're paying attention, the `GD_SHARED_LIBADD -> GDLIB_LIBS` change was made under the assumption that `PHP_GD` is essentially binary.